### PR TITLE
Add range context support for media-feature-name-* rules

### LINKED ISSUE #1744: Add range context support to media-feature-name-* rules
Ref: https://github.com/stylelint/stylelint/pull/1732#issuecomment-235888292

``` css
@media (10px < width > 50px) {}
```

Is dependent on the creation of a [range context](https://www.w3.org/TR/mediaqueries-4/#mq-range-context) parser.

### PR DESCRIPTION
**Pull request by jdiogopeixoto**
> Which issue, if any, is this issue related to?

Closes #1744 

> Is there anything in the PR that needs further explanation?

No, it's self explanatory.

### CODE REVIEW

File: lib/rules/rangeContextNodeParser.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+		operators.push(match.target);
+	});
+
+	// Sorting helps when using the operators to split
+	// E.g. for "10em < width <= 50em" this returns ["<=", "<"]
+	return operators.sort((a, b) => b.length - a.length);
+}
+
+function getRangeContextName(parsedNode) {
+	if (parsedNode.length === 3) {
```
Comment (hudochenkov) [inline lib/rules/rangeContextNodeParser.js]: Could you add a comment into the code explaining what is this `3`? It's not clear where this number came from.

Comment (jdiogopeixoto) [reply lib/rules/rangeContextNodeParser.js]: Yes, I added more context to that.

**Comment by jeddy3** [comment]
I fixed the conflict from merging https://github.com/stylelint/stylelint/pull/4561.

**Comment by jeddy3** [comment]
After we merge this, I work on https://github.com/stylelint/stylelint/issues/4539 as I believe this is the last unmerged pull request with documentation changes.

**Comment by jeddy3** [comment]
> -   Fixed: `media-feature-name-*` false negatives for range context ([#4581](https://github.com/stylelint/stylelint/pull/4581)).

**Pull request merged**